js/main.js

```diff
@@ -317,26 +327,29 @@ function cityIdToLabel(s) {
 /*
  * Set the current city.
  *
- * `city` is the city ID.
+ * `cityID` is the new city's ID. If `newHistoryState` is true then a new
+ * entry in the browser's history is created for the change. Otherwise the
+ * current history entry is replaced.
  */
-function setCity(city) {
-    var filename = 'cities/' + city + '.json';
+function setCity(cityID, newHistoryState) {
```

Comment:
@torfsen You can hide the boolean in a separate function:

``` js
function setCityWithNewHistoryEntry(cityID) {
    setCity(cityID, true);
}
```

and use this function whenever a history entry has to be created. This will also make the comments redundant since the function name describes the intended behavior.

For the `false` case I would check within `setCity()` if the parameter is set:

``` js
function setCity(cityID, createNewHistoryEntry) {
    createNewHistoryEntry = createNewHistoryEntry || false;
    // ...
```

Then all invocations outside of `setCity()` can be simplified to using one parameter as in `setCity(getHashCity());`